### PR TITLE
[Ag Grid Theme] Ag grid v30 - no sass

### PR DESCRIPTION
**Pull request by alycrys**
Closes https://github.com/jpmorganchase/salt-ds/issues/2602

Storybook Preview: https://github.com/jpmorganchase/salt-ds/pull/2661#issuecomment-1787072993

### CODE REVIEW

File: packages/ag-grid-theme/stories/ag-grid-theme.stories.tsx [hunk 1]
```diff
@@ -1,8 +1,7 @@
 import { AgGridReact } from "ag-grid-react";
-import "ag-grid-community/dist/styles/ag-grid.css";
-import "ag-grid-community/dist/styles/ag-theme-material.css";
-import "../uitk-ag-theme.css";
-import "../salt-ag-theme.css";
+import "ag-grid-community/styles/ag-grid.css";
+import "ag-grid-community/styles/ag-theme-material.css";
```
Comment (origami-z) [inline packages/ag-grid-theme/stories/ag-grid-theme.stories.tsx:3]: We don't need ag grid material style?

File: packages/ag-grid-theme/stories/ag-grid-theme.stories.tsx [hunk 2]
```diff
@@ -1,8 +1,7 @@
 import { AgGridReact } from "ag-grid-react";
-import "ag-grid-community/dist/styles/ag-grid.css";
-import "ag-grid-community/dist/styles/ag-theme-material.css";
-import "../uitk-ag-theme.css";
-import "../salt-ag-theme.css";
+import "ag-grid-community/styles/ag-grid.css";
```
Comment (origami-z) [inline packages/ag-grid-theme/stories/ag-grid-theme.stories.tsx:2]: Is this needed? if so, usage doc needs update

File: .github/workflows/chromatic.yml [hunk 3]
```diff
@@ -23,8 +23,6 @@ jobs:
           cache: "yarn"
       - name: Install dependencies
         run: yarn
-      - name: Build ag grid theme
```
Comment (origami-z) [inline .github/workflows/chromatic.yml:26]: Why remove?

File: packages/ag-grid-theme/package.json [hunk 4]
```diff
@@ -14,8 +14,7 @@
     "/salt-ag-theme.css"
   ],
   "scripts": {
-    "build": "yarn node ./scripts/build.mjs",
-    "build-watch": "yarn node ./scripts/build.mjs --watch"
```
Comment (origami-z) [inline packages/ag-grid-theme/package.json:18]: Probably worth to have a watch for easy dev?

**Review by origami-z — COMMENTED** [review]
Can we not rely on material theme? That feels we're going back towards mui...

- editable cell has some diff
- filter input has some diff, vertical alignment

File: packages/ag-grid-theme/stories/ag-grid-theme.stories.tsx (hunk 1 above)
Comment (alycrys) [reply packages/ag-grid-theme/stories/ag-grid-theme.stories.tsx:3]: There are some icons that we don't provide (ie. the grip icon in DragRowExample)

File: packages/ag-grid-theme/stories/ag-grid-theme.stories.tsx (hunk 2 above)
Comment (alycrys) [reply packages/ag-grid-theme/stories/ag-grid-theme.stories.tsx:2]: Yes. We apply our styles on top of the ag grid styles. Will update docs in the next PR, there will be a lot of changes there

File: .github/workflows/chromatic.yml (hunk 3 above)
Comment (alycrys) [reply .github/workflows/chromatic.yml:26]: no need, as we're not using sass anymore

File: packages/ag-grid-theme/stories/ag-grid-theme.stories.tsx (hunk 1 above)
Comment (origami-z) [reply packages/ag-grid-theme/stories/ag-grid-theme.stories.tsx:3]: How does the icon work currently? Is it from Sass?

We don't really want go backwards being seen as built on top of other theme, unless no other option is there.

Comment (alycrys) [reply packages/ag-grid-theme/stories/ag-grid-theme.stories.tsx:3]: the icons come from the theme fonts. no sass involved

Comment (origami-z) [reply packages/ag-grid-theme/stories/ag-grid-theme.stories.tsx:3]: If icon is the main problem, let's ask @pseys on suggested icons. Some were added last time as well, e.g. group

**Comment by alycrys** [comment]
> Can we not rely on material theme? That feels we're going back towards mui...
> 
> * editable cell has some diff
> * filter input has some diff, vertical alignment

The diffs have been confirmed and agreed by design

File: packages/ag-grid-theme/package.json (hunk 4 above)
Comment (alycrys) [reply packages/ag-grid-theme/package.json]: This is not helpful as we're not using sass anymore

File: packages/ag-grid-theme/package.json [hunk 5]
```diff
@@ -20,14 +20,5 @@
   "publishConfig": {
     "directory": "../../dist/salt-ds-ag-grid-theme",
     "provenance": true
-  },
-  "devDependencies": {
```
[the remark this answers is not in the document]
Comment (alycrys) [reply packages/ag-grid-theme/package.json:24]: You're right. I missed that. Thanks

File: packages/ag-grid-theme/css/uitk-icons.css [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+  font-family: "uitk-icons";
+  content: "\e965";
+}
+
+[class*="ag-theme-uitk"] .ag-icon-filter:before {
+  font-family: "uitk-icons";
+  content: "\e922";
+}
+
+/* [class*="ag-theme-uitk"] .ag-icon-filter-clear:before {
```
Comment (origami-z) [inline packages/ag-grid-theme/css/uitk-icons.css:91]: Missing?

File: packages/ag-grid-theme/src/dependencies/useAgGridHelpers.ts [hunk 7]
```diff
@@ -60,7 +60,7 @@ export function useAgGridHelpers({
     }
   }, [density, agThemeName, compact]);
 
-  const className = `${agThemeName}-${density}${
+  const className = `${agThemeName}${
```
Comment (origami-z) [inline packages/ag-grid-theme/src/dependencies/useAgGridHelpers.ts:63]: Worth calling out in noticeable places

**Review by origami-z — APPROVED** [review]

**Comment by joshwooding** [comment]
The vercel build fails with:

image

File: packages/ag-grid-theme/src/dependencies/useAgGridHelpers.ts (hunk 7 above)
Comment (alycrys) [reply packages/ag-grid-theme/src/dependencies/useAgGridHelpers.ts:63]: not sure I understand

File: packages/ag-grid-theme/css/uitk-icons.css (hunk 6 above)
Comment (alycrys) [reply packages/ag-grid-theme/css/uitk-icons.css:91]: yes. Also waiting for design to provide a grab icon (for the drag & drop rows)

**Comment by alycrys** [comment]
> The vercel build fails with:
> 
> image

sorted. thanks

**Pull request merged**